validator/validators/performance/performance_validator.go

```diff
@@ -0,0 +1,239 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package performance
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/validator/models"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/cenkalti/backoff/v4"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+)
+
+const (
+	ServiceName      = "AmazonCloudWatchAgent"
+	DynamoDBDataBase = "CWAPerformanceMetrics"
+)
+
+var (
+	metricsConvertToMB = []string{"mem_total", "procstat_memory_rss", "procstat_memory_swap", "procstat_memory_data", "procstat_memory_vms", "procstat_write_bytes", "procstat_bytes_sent"}
+)
+
+type PerformanceValidator struct {
+	vConfig models.ValidateConfig
+}
+
+var _ models.ValidatorFactory = (*PerformanceValidator)(nil)
+
+func NewPerformanceValidator(vConfig models.ValidateConfig) models.ValidatorFactory {
+	return &PerformanceValidator{
+		vConfig: vConfig,
+	}
+}
+
+func (s *PerformanceValidator) GenerateLoad() (err error) {
+	var (
+		agentCollectionPeriod = s.vConfig.GetAgentCollectionPeriod()
+		agentConfigFilePath   = s.vConfig.GetCloudWatchAgentConfigPath()
+		dataType              = s.vConfig.GetDataType()
+		dataRate              = s.vConfig.GetDataRate()
+		receiver              = s.vConfig.GetPluginsConfig()
+	)
+	switch dataType {
+	case "logs":
+		err = common.StartLogWrite(agentConfigFilePath, agentCollectionPeriod, dataRate)
+	default:
+		err = common.StartSendingMetrics(receiver, agentCollectionPeriod, dataRate)
+	}
+
+	return err
+}
+
+func (s *PerformanceValidator) CheckData(startTime, endTime time.Time) error {
+	metrics, err := s.GetPerformanceMetrics(startTime, endTime)
+	if err != nil {
+		return err
+	}
+
+	perfInfo, err := s.CalculateMetricStatsAndPackMetrics(metrics.MetricDataResults)
+	if err != nil {
+		return err
+	}
+
+	err = s.SendPacketToDatabase(perfInfo)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *PerformanceValidator) Cleanup() error {
+	var (
+		dataType      = s.vConfig.GetDataType()
+		ec2InstanceId = awsservice.GetInstanceId()
+	)
+	switch dataType {
+	case "logs":
+		awsservice.DeleteLogGroup(ec2InstanceId)
+	}
+
+	return nil
+}
+
+func (s *PerformanceValidator) SendPacketToDatabase(perfInfo PerformanceInformation) error {
```

Comment:
Similar to [update item ](https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/test/performancetest/transmitter.go#L188-L254)with reduced code. Basically Get the latest performance information from the database and update by merging the existing one and finally replace the packet in the database if exist. If the package is not exist, we will create a new one in the database.